pandas/io/pytables.py

```diff
@@ -4473,7 +4473,7 @@ def _reindex_axis(obj, axis, labels, other=None):
 
     labels = ensure_index(labels.unique())
     if other is not None:
-        labels = ensure_index(other.unique()) & labels
+        labels = ensure_index(other.unique()).intersection(labels, sort=False)
```

Comment:
Again, on Master ``intersection`` does not generally sort by default so I have added ``sort=False`` here for compatibility